src/vault/client/ext/aws.clj

```diff
@@ -41,13 +44,15 @@
     (.setContent (-> payload (.getBytes "UTF-8") (io/input-stream)))))
 
 
-(defn credentials
+(defn derive-credentials
   "AWSCredentials to authenticate with STS. Will fall back to default chain including instance profile."
   ^AWSCredentials
   ([id secret token]
    (BasicSessionCredentials. id secret token))
+  ([id secret]
+   (BasicAWSCredentials. id secret))
   ([]
-   (-> (DefaultAWSCredentialsProviderChain/getInstance)
+   (-> (DefaultAWSCredentialsProviderChain.)
```

Comment:
Hmm, why this change? Maybe because we mess with the default singleton in our services? 🤔